src/Queue/Management/Task.php

```diff
@@ -0,0 +1,176 @@
+        $currentStatus = $request->postVar('currentStatus');
+        $status = $request->postVar('status');
+
+        if ($implementation
+            && $status
+            && ($implementation === 'all' || in_array($implementation, $implementations))
+            && ($currentStatus === 'any' || in_array($currentStatus, $currentStatuses))
+            && in_array($status, $statuses)
+        ) {
+            $where = [
+                ['`JobStatus` != ?' => QueuedJob::STATUS_COMPLETE],
+            ];
+
+            // Filter by implementation
+            $where[] = $implementation === 'all'
+                ? '`Implementation` IN ' . sprintf(
+                    "('%s')",
+                    str_replace('\\', '\\\\', implode("','", $implementations))
+                )
+                : ['`Implementation`' => $implementation];
+
+            // Filter by status
+            if ($currentStatus !== 'any') {
+                $where[] = ['`JobStatus`' => $currentStatus];
+            }
+
+            // Assemble query
+            $query = SQLUpdate::create(
+                'QueuedJobDescriptor',
+                [
+                    'JobStatus' => $status,
+                    // make sure to reset all data which is related to job management
+                    // job lock
+                    'Worker' => null,
+                    'Expiry' => null,
+                    // resume / pause
+                    'ResumeCounts' => 0,
+                    // broken job notification
+                    'NotifiedBroken' => 0,
+                ],
+                $where
+            );
+
+            $query->execute();
+
+            echo sprintf('Job status updated (%d rows affected).', DB::affected_rows());
+
+            return;
+        }
+
+        echo '<form action="" method="post">';
+
+        echo '<p>Job type</p>';
+
+        echo '<select name="implementation">';
+        echo '<option value="all">All job types</option>';
+
+        foreach ($implementations as $item) {
+            echo sprintf('<option value="%s">%s</option>', $item, $item);
+        }
+
+        echo '</select>';
+
+        echo '<br />';
+
+        echo '<p>Current status</p>';
+
+        echo '<select name="currentStatus">';
+        echo '<option value="any">Any status (except Complete)</option>';
+
+        foreach ($currentStatuses as $item) {
+            echo sprintf('<option value="%s">%s</option>', $item, $item);
+        }
+
+        echo '</select>';
+
+        echo '<br />';
+
+        echo '<p>Update status to</p>';
+
+        echo '<select name="status">';
+
+        foreach ($statuses as $item) {
+            echo sprintf('<option value="%s">%s</option>', $item, $item);
+        }
+
+        echo '</select>';
+
+        echo '<br />';
+
+        echo '<br />';
+
+        echo 'Submitting will apply change immediately:<br><br>';
+        echo '<button type="submit">Update status</button>';
```

Comment:
While the `TaskRunner` controller enforces authorisation on this task (effectively CLI or ADMIN), it's not automatically guarding against CSRF. While CSRF via POST is a bit harder to achieve, this would still make logged-in admins vulnerable. Check out `URLSpecialsMiddleware` in core in how we've handled that for `dev/build`, you might be able to hook into the same system? Alternatively, use the Silverstripe `Form` APIs which should enable CSRF by default.

Reply:
I agree, I was hoping to use forms and templates to improve this dev task :)